app/components/AccountConnection.jsx

```diff
@@ -1,62 +1,81 @@
 /** @jsx h */
-import { h } from 'preact'
+import { h, Component } from 'preact'
 import statefulForm from '../lib/statefulForm'
 
 import AccountLoginForm from './AccountLoginForm'
 import DataItem from './DataItem'
 import ReactMarkdown from 'react-markdown'
 
-const AccountConnection = ({ t, connector, fields, dirty, error, submit, submitting }) => {
-  const { name, customView, description } = connector
-  // If there is no field displayed, the form is dirty by default.
-  dirty = dirty || Object.values(fields).every(field => field.type === 'hidden' || field.advanced)
-  return (
-    <div class='account-connection'>
-      <div class='account-description'>
-        <h3>{t('my_accounts title description')}</h3>
-        <p>
-          <ReactMarkdown
-            source={
-              t(description)
-            }
-            renderers={{Link: props => <a href={props.href} target='_blank'>{props.children}</a>}}
-          />
-        </p>
-        <h3>{t('dataType title')}</h3>
-        <ul class='account-datas'>
-          {connector.dataType && connector.dataType.map(data =>
-            <DataItem
-              dataType={data}
-              hex={connector.color.hex}
+class AccountConnection extends Component {
+  constructor (props, context) {
+    super(props, context)
+
+    this.state = {
+      submitting: false
+    }
+
+    this.submit = () => {
+      this.setState({
+        submitting: true
+      })
+
+      this.props.submit()
+    }
+  }
+
+  render ({ t, connector, fields, dirty, error, submit }) {
```

Comment:
Be careful here, Preact provide props in the render here but it's not the case for React. Usually we grab props after the definition with `this.props` (also my preference).